File: towards/towards.go
package towards

var (
	cache = map[int][]int{}
)

func Towards1(n int) (a []int) {
	for n > 1 {
		a = append(a, n)
		if n%2 == 0 {
			n = n / 2
		} else {
			n = n*3 + 1
		}
	}
	a = append(a, n)
	return
}

File: towards/towards.jl
#!/usr/bin/env julia

function Towards1(n::Int64)
    a = Int64[]
    while n > 1
        push!(a, n)
        if n % 2 == 0
            n = n/2
        else
            n = n*3 + 1
        end
    end
    push!(a,n)
    return a
end
